### PR TITLE
intergration test scaffolding, converted to es6 modules

### PR DESCRIPTION
**Pull request by jmartin84**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ require('babel/register')({
 var jsPaths = ['client/**/*.js', 'server/**/*.js'];
 
 gulp.task('test', function () {
-    return gulp.src(jsPaths, {read:false})
+    return gulp.src(jsPaths, {read: false})
```
Comment (CWSpear) [inline gulpfile.js]: I thought we all agreed we prefer `{ read: false }` spacing-wise.

File: server/index.js [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
 
-var loopback = require('loopback');
-var boot = require('loopback-boot');
-var debug = require('debug')('server');
+var app = loopback();
 
-var server = loopback();
-
-server.start = () => {
+app.start = function() {
```
Comment (CWSpear) [inline server/index.js]: Why did you move away from `es6`? (Particularly curious when "converted to es6" is in the title of the PR!)

File: server/model-config.json [hunk 3]
```diff
@@ -14,22 +14,26 @@
...
-    "dataSource": "db",
+    "dataSource": "pg",
     "public": false
   },
   "Role": {
-    "dataSource": "db",
+    "dataSource": "pg",
     "public": false
+  },
+  "paste": {
```
Comment (CWSpear) [inline server/model-config.json:35]: The other keys have Uppercase, but this one is lowercase. Any particular reason.

Do we have to use `json` here? (I assume this is a LoopBack config thing?)

File: tests/integration/paste.test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            })
+            .expect(201)
+            .expect(function(res) {
+                res.body.should.have.property('slug').and.exist();
+                res.body.should.have.property('createdById').and.exist();
+            });
+    });
+
+    it('should return 500 when post fails', () => {
+        true.should.be.false;
```
Comment (CWSpear) [inline tests/integration/paste.test.js:32]: This always fails, too, then?

File: tests/integration/server.test.js [hunk 5]
```diff
@@ -1,28 +1,14 @@
...
-    beforeEach(() => {
-        app = server.start();
-    });
+describe('INTEGRATION-TESTS::Server tests', () => {
+    beforeEach(() => app = server.start());
 
-    afterEach((done) => {
-        app.close(done);
-    });
+    afterEach((done) => app.close(done));
```
Comment (CWSpear) [inline tests/integration/server.test.js]: Does `app.close()` not return a promise?

Comment (CWSpear) [inline tests/integration/server.test.js]: Code-style-wise, I like leaving off the parentheses around the `done` param there (single param, very unlikely to have a second param, etc).

File: gulpfile.js (hunk 1 above)
Comment (jmartin84) [inline gulpfile.js]: Yeah I forgot what we settled on, you wanna set it up in eslint?

File: server/index.js (hunk 2 above)
Comment (jmartin84) [inline server/index.js]: I reran the yeoman generator and didn't really go through this file but mostly what I did when I say 'converted to es6' is that I moved to the es6 modules instead of commonJS

File: server/model-config.json (hunk 3 above)
Comment (jmartin84) [inline server/model-config.json:35]: the upper case ones are buildin loopback things I haven't touched, but I'm probably going to make those lowercase since it doesn't seem that you ever new the models - but we will see what we see.

Also yes is a loopback config thing, you can change the models programatically with JS but the model generator give the json and they seem to want you to use the JS only to override methods

File: tests/integration/paste.test.js (hunk 4 above)
Comment (jmartin84) [inline tests/integration/paste.test.js:32]: all these tests are skipped, but where you see true === false its because they obviously aren't implemented. I usually put in some forced assert fail so that they stand out during development.

File: tests/integration/server.test.js (hunk 5 above)
Comment (jmartin84) [inline tests/integration/server.test.js]: app close does not return a promise and I didn't feel the need to promisify it.

as for () => I just thought the parens were required.

**Comment by CWSpear** [comment]
Are you expecting the tests to pass at this point?

**Comment by jmartin84** [comment]
Yeah, I'm pretty close I just haven't gotten back to them since last week.
